File: 283-move-zeroes/move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let NonZeroIndex=0;
    for(let i=0;i<nums.length;i++){
    if(nums[i]!=0){
        nums[NonZeroIndex]=nums[i];
        NonZeroIndex++;
    }
    }
    for(let i=NonZeroIndex;i<nums.length;i++){
    nums[i]=0;
    }
};

File: 283-move-zeroes/move-zeroes.go
func moveZeroes(nums []int)  {
    nonZero:=0
    for i:=0;i<len(nums);i++{
        if nums[i]!=0{
            nums[nonZero]=nums[i]
            nonZero++
        }
    }
    for nonZero<len(nums){
        nums[nonZero]=0
        nonZero++
    }
}